dev/benchmarks/microbenchmarks/lib/geometry/matrix_utils_transform_rect_bench.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../common.dart';
+
+const int _kNumIterations = 1000000;
+const int _kNumWarmUp = 10000;
+
+void main() {
+  assert(false, "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
```

Comment:
Let's print a warning here instead of `assert(false)` in the checked/debug mode. Sometimes, we would intentionally run the test in debug mode to either debug the code, or check that there's no unexpected error in the code.

You can make the warning message standout using a format like https://github.com/flutter/flutter/blob/master/packages/flutter_driver/lib/src/driver/driver.dart#L147

Reply:
This same line appears in every microbenchmark. Yegor included it in his original PR and I kept it.

Anecdotally, it actually stopped me a couple of times when I forgot to run the benchmark with --release so it was worthwhile.

If you think this pattern should be updated, perhaps that should be a separate issue/PR?